From: Dave Jiang <dave.jiang@intel.com>
Date: Wed, 1 Jun 2011 09:03:08 +0000
Subject: [PATCH 227/273] isci: Removing unused variables compiler warnings
Git-commit: 8d2c65c09c9e0adc16070562e7944c1c3277f332
References: FATE#311808,bnc#709528
Patch-Mainline: 3.0

Newer gcc's are better at identifying "set, but not used" variables.

Signed-off-by: Dave Jiang <dave.jiang@intel.com>
Signed-off-by: Dan Williams <dan.j.williams@intel.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/isci/request.c |    8 --------
 drivers/scsi/isci/task.c    |    3 +--
 2 files changed, 1 insertions(+), 10 deletions(-)

diff --git a/drivers/scsi/isci/request.c b/drivers/scsi/isci/request.c
index 7c0928e..31c9b2c 100644
--- a/drivers/scsi/isci/request.c
+++ b/drivers/scsi/isci/request.c
@@ -211,11 +211,9 @@ static void scu_ssp_reqeust_construct_task_context(
 	struct scu_task_context *task_context)
 {
 	dma_addr_t dma_addr;
-	struct scic_sds_controller *controller;
 	struct scic_sds_remote_device *target_device;
 	struct scic_sds_port *target_port;
 
-	controller = scic_sds_request_get_controller(sds_request);
 	target_device = scic_sds_request_get_device(sds_request);
 	target_port = scic_sds_request_get_port(sds_request);
 
@@ -384,11 +382,9 @@ static void scu_sata_reqeust_construct_task_context(
 	struct scu_task_context *task_context)
 {
 	dma_addr_t dma_addr;
-	struct scic_sds_controller *controller;
 	struct scic_sds_remote_device *target_device;
 	struct scic_sds_port *target_port;
 
-	controller = scic_sds_request_get_controller(sci_req);
 	target_device = scic_sds_request_get_device(sci_req);
 	target_port = scic_sds_request_get_port(sci_req);
 
@@ -677,12 +673,10 @@ enum sci_status scic_task_request_construct_ssp(
 static enum sci_status scic_io_request_construct_basic_sata(struct scic_sds_request *sci_req)
 {
 	enum sci_status status;
-	struct scic_sds_stp_request *stp_req;
 	bool copy = false;
 	struct isci_request *isci_request = sci_req_to_ireq(sci_req);
 	struct sas_task *task = isci_request_access_task(isci_request);
 
-	stp_req = &sci_req->stp.req;
 	sci_req->protocol = SCIC_STP_PROTOCOL;
 
 	copy = (task->data_dir == DMA_NONE) ? false : true;
@@ -3190,7 +3184,6 @@ scu_smp_request_construct_task_context(struct scic_sds_request *sci_req,
 				       ssize_t req_len)
 {
 	dma_addr_t dma_addr;
-	struct scic_sds_controller *scic;
 	struct scic_sds_remote_device *sci_dev;
 	struct scic_sds_port *sci_port;
 	struct scu_task_context *task_context;
@@ -3202,7 +3195,6 @@ scu_smp_request_construct_task_context(struct scic_sds_request *sci_req,
 
 	task_context = scic_sds_request_get_task_context(sci_req);
 
-	scic = scic_sds_request_get_controller(sci_req);
 	sci_dev = scic_sds_request_get_device(sci_req);
 	sci_port = scic_sds_request_get_port(sci_req);
 
diff --git a/drivers/scsi/isci/task.c b/drivers/scsi/isci/task.c
index 93c7556..b54ef2b 100644
--- a/drivers/scsi/isci/task.c
+++ b/drivers/scsi/isci/task.c
@@ -1358,7 +1358,6 @@ isci_task_request_complete(struct isci_host *ihost,
 			   enum sci_task_status completion_status)
 {
 	struct isci_remote_device *idev = ireq->isci_device;
-	enum isci_request_status old_state;
 	struct isci_tmf *tmf = isci_request_access_tmf(ireq);
 	struct completion *tmf_complete;
 	struct scic_sds_request *sci_req = &ireq->sci;
@@ -1367,7 +1366,7 @@ isci_task_request_complete(struct isci_host *ihost,
 		"%s: request = %p, status=%d\n",
 		__func__, ireq, completion_status);
 
-	old_state = isci_request_change_state(ireq, completed);
+	isci_request_change_state(ireq, completed);
 
 	tmf->status = completion_status;
 	ireq->complete_in_target = true;
-- 
1.6.0.2

